### ISSUE TITLE
Popover not closing on click-outside when used in Modal

### ISSUE THREAD

**Issue by HardeepAsrani** [WordPress/gutenberg#21370]
**Describe the bug**
If you use the Popover component or anything that uses the Popover component, such as ColorPalette, in Modal component then it doesn't close once you click outside.

**To reproduce**
1. Add ColorPalette control to Modal.
2. Click on Custom color which opens color-picker popover.
3. Try clicking outside and it won't close.

**Expected behavior**
Like in the rest of the editor, it should close.

**Screenshots**
Here's a screencast of the issue: https://www.loom.com/share/bcd8bb35dac5445c967b2d6708f6ceb9

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Firefox
 - Version 74.0

**Additional context**
- Gutenberg 7.8.1

**Comment by HardeepAsrani** [comment]
If you want to test it pretty quickly, use this:

```
( function( blocks, element, components ) {
	var el = element.createElement;
	var Modal = components.Modal;
	var ColorPalette = components.ColorPalette;

	blocks.registerBlockType( 'gutenberg-examples/example-01-basic', {
		title: 'Example: Basic',
		icon: 'universal-access-alt',
		category: 'layout',
		example: {},
		edit: function() {
			return el(
				Modal,
				{ title: 'Hello World' },
				el(
					ColorPalette,
					{
						value: '#ffffff',
						onClick: e => console.log( e )
					}
				)
			);
		},
		save: function() {
			return el(
				'p',
				'Hello World, step 1 (from the frontend).'
			);
		}
	});
}(
	window.wp.blocks,
	window.wp.element,
	window.wp.components,
	window.wp.blockEditor
) );
```

Of course, it's a bad example but it shows the issue. :)

**Comment by tellthemachines** [comment]
Can confirm by using the above snippet that a popover inside a modal won't close when the click outside happens inside the modal. The popover does close if the click outside happens outside of the modal, but only if the modal hasn't been clicked yet.

**Comment by stokesman** [comment]
I tested this using the snippet provided by @HardeepAsrani and this seems to no longer be an issue. The popover closes no matter where I click. 

https://user-images.githubusercontent.com/9000376/171073768-2fb5f1c5-ef34-4e32-a438-8138c0549913.mp4

Tested with Gutenberg as running at wordpress.org/gutenberg on macOS/Firefox 100.

I'm closing this but let me know if I missed something and I'll reopen.
